src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -297,7 +303,14 @@ def get_request_status(
     else:
         PrivacyRequest.execution_logs_by_dataset = property(lambda self: None)
 
-    return paginate(query, params)
+    paginated = paginate(query, params)
+    if include_identities:
+        # Conditionally include the cached identity data in the response if
+        # it is explicitly requested
+        for item in paginated.items:
+            item.identity = item.get_cached_identity_data()
```

Comment:
`paginate` also loops through every the every element in the result set before creating the page, and then we are turning around and looping through them again.


```python
def paginate(query: Query, params: Optional[AbstractParams] = None) -> AbstractPage:
    params = resolve_params(params)

    total = query.count()
    items = [_to_dict(item) for item in paginate_query(query, params)]

    return create_page(items, total, params)
```

But we are just looping through fifty elements twice, so not a big deal. Another option is having a custom paginate method here that is similar to the above that also adds the identity data, killing two birds.

Reply:
IMO it's best we keep these separate so it's clear we're adding this data outside of the scope of our DB and pagination functionality (given it's a very small number of iterations). Where this could prove problematic is with the CSV download functionality, where pagination isn't definitely applied.

Reply:
sounds good